store/index.js

```diff
@@ -388,63 +410,12 @@ function toAd(i, r) {
   }
 }
 
-function eventToAd(state, adEvent) {
-  let ad = {
-    idx: adEvent.idx.toNumber !== undefined ? adEvent.idx.toNumber() : adEvent.idx
-  }
+function appendAd(state, rawAd) {
+  let ad = normalizeAd(rawAd);
 
-  if (adEvent.owner !== undefined) {
-    ad.owner = normalizeAddr(adEvent.owner);
-  }
-  if (adEvent.to !== undefined) { // Convert SetAdOwner event
-    ad.owner = normalizeAddr(adEvent.to);
-  }
-  if (adEvent.x !== undefined) { // Normalize ints for Buy event
-    ad.x = Number(adEvent.x);
-    ad.y = Number(adEvent.y);
-    ad.width = Number(adEvent.width);
-    ad.height = Number(adEvent.height);
-  }
-  if (adEvent.link !== undefined) {
-    ad.link = adEvent.link;
-    ad.image = adEvent.image;
-    ad.title = adEvent.title;
-  }
-  if (adEvent.forceNSFW === true) { // Force NSFW
-    ad.NSFW = true;
-  } else if (adEvent.NSFW !== undefined) {
-    ad.NSFW = adEvent.NSFW;
-  }
-  // TODO will this break if we load stuff from events and not KetherView?
-  if (adEvent.wrapped !== undefined) {
-    ad.wrapped = adEvent.wrapped;
-  } else {
-    ad.wrapped = normalizeAddr(adEvent.owner) === state.networkConfig.ketherNFTAddr;
-  }
-  let existingAd = state.ads[ad.idx];
-  if (existingAd !== undefined && existingAd.width !== undefined) {
-    // Already counted, update values
-    return Object.assign({}, existingAd, ad);
-  }
-
-  // Add defaults to non-existing ad
-  return Object.assign({
-    link: "",
-    image: "",
-    title: "",
-    NSFW: false,
-    forceNSFW: false,
-    wrapped: false,
-  }, ad);
-}
-
-function appendAd(state, ad) {
   if (state.accounts[ad.owner]) {
     addAdOwned.call(this, state, ad);
-  } else if (ad.owner === state.networkConfig.ketherNFTAddr) {
```

Comment:
Was this removed intentionally? This was the fix to #94

Reply:
Yes. InitState now get this thru Transfer events.